src/db/collections/ledger_update.rs

```diff
@@ -119,19 +119,31 @@ impl MongoDb {
         // TODO: Use `insert_many` and `update_many` to increase write performance.
 
         for OutputWithMetadata { output, metadata } in outputs_with_metadata {
+            let at = metadata.spent.map_or(metadata.booked, |s| s.spent);
+            let is_spent = metadata.spent.is_some();
+
             // Ledger updates
             for owner in output.owning_addresses() {
                 let ledger_update_document = LedgerUpdateDocument {
                     address: owner,
                     output_id: metadata.output_id,
-                    at: metadata.spent.map_or(metadata.booked, |s| s.spent),
-                    is_spent: metadata.spent.is_some(),
+                    at,
+                    is_spent,
                 };
 
+                let mut doc = bson::to_document(&ledger_update_document)?;
+                doc.insert("_id", ledger_update_document.output_id.to_hex());
```

Comment:
The `outputId` is not unique in this collection, because there can be the same output id for spent and unspent outputs.